batch/batch/cloud/azure/worker/disk.py

```diff
@@ -1,27 +1,185 @@
 
-    async def create(self, labels=None):
-        raise NotImplementedError
+                delay = 1
+                while True:
+                    spec = await self.compute_client.get(f'/virtualMachines/{self.instance_name}/instanceView')
+                    disk_status = self._extract_disk_from_disks(spec['disks'])
+                    if disk_status is None:
+                        self.attached = False
+                        return
+                    delay = await sleep_and_backoff(delay, max_delay=10)
+
+    async def _delete(self):
+        async with LoggingTimer(f'deleting disk {self.name}'):
+            if self.created:
+                try:
+                    params = {'api-version': '2020-12-01'}
+                    await self.compute_client.delete(f'/disks/{self.name}', params=params)
+                except aiohttp.ClientResponseError as e:
+                    if e.status in (204, 404):  # https://docs.microsoft.com/en-us/rest/api/compute/disks/delete#response
+                        pass
+
+                self.created = False
+
+    async def create(self, labels: Optional[Dict[str, str]] = None):
+        del labels  # disks automatically inherit the VM's labels
+        await self._create_and_attach()
+        await self._format()
 
     async def delete(self):
-        raise NotImplementedError
+        try:
+            await self._unmount()
+        finally:
+            try:
+                await self._detach()
+            finally:
+                await self._delete()
 
     async def close(self):
-        raise NotImplementedError
+        await self.compute_client.close()
+
+
+class AzureDiskManager(CloudDiskManager):
+    def __init__(self, machine_name: str, instance_config: AzureSlimInstanceConfig, subscription_id: str, resource_group: str):
+        self.subscription_id = subscription_id
+        self.resource_group = resource_group
+
+        self.disk_mapping: Dict[str, dict] = {config['name']: config for config in instance_config.data_disks(machine_name)}
+
+        self.update_lock = asyncio.Lock()
+        self.lun_queue: asyncio.Queue[int] = asyncio.Queue()
+
+        start_lun = 0 if instance_config.local_ssd_data_disk else 1  # FIXME: double check this works!
+        max_lun = min(32, instance_config.cores * 2)  # https://docs.microsoft.com/en-us/azure/virtual-machines/ddv4-ddsv4-series
+        for lun in range(start_lun, max_lun):
+            self.lun_queue.put_nowait(lun)
+
+    def vm_disk_config(self):
+        return {
+            'properties': {
+                'storageProfile': {
+                    'dataDisks': list(self.disk_mapping.values())
+                }
+            }
+        }
+
+    def _add_disk(self, disk: AzureDisk) -> dict:
+        self.disk_mapping[disk.name] = disk.spec()
+        return self.vm_disk_config()
+
+    def _remove_disk(self, disk: AzureDisk) -> dict:
+        self.disk_mapping.pop(disk.name)
```

Comment:
`del self.disk_mapping[disk.name]` should have the same semantics and has pleasing symmetry with line 164.